tests/foreman/cli/test_user.py

```diff
@@ -295,20 +214,22 @@ def test_negative_create_with_invalid_username(self):
 
         @Assert: User is not created. Appropriate error shown.
         """
-        for invalid_name in ('',
-                             'space {0}'.format(gen_string('alpha')),
-                             gen_string('alpha', 101),
-                             gen_string('html')):
-            with self.subTest(invalid_name):
+        invalid_names = (
+            '', 'space {0}'.format(gen_string('alpha')),
+            gen_string('alpha', 101),
+            gen_string('html')
+        )
+        for invalid_name in invalid_names:
+            with self.subTest(invalid_name), self.assertRaises(
+                    CLIReturnCodeError):
```

Comment:
assertRaises will just look for the specified exception and will raise AssertionError depending if the exception was not raised. So it won't catch any other exception.

I prefer to just wrap with the assertRaises just the statements we are expecting to fail, that makes the code more clear on where we expect the exception to happen. Using nested context manager is good for decreasing the indentation level, but I think having the meaning of where we expect a failure wins.